foundation/otel/sampler.go

```diff
@@ -16,14 +16,31 @@ func newEndpointExcluder(endpoints map[string]struct{}, probability float64) end
 	}
 }
 
+func endpoint(parameters trace.SamplingParameters) string {
+	var path, query string
+
+	for _, attr := range parameters.Attributes {
+		switch attr.Key {
+		case "url.path":
+			path = attr.Value.AsString()
+		case "url.query":
+			query = attr.Value.AsString()
+		}
+	}
+
+	if path != "" && query != "" {
+		path += "?" + query
```

Comment:
We should return the value here and then return "" at the end. When you write code like this where the return is not explict, it requires a lot of mental effort to understand what is going on.

I don't remember any of this code :(

Reply:
I looked more and I see we are using it to remove liveness and readiness from the traces. So you are getting the route and then checking it. I can test this, but lets clean up the code first.